### ISSUE TITLE
Matching tuples using wildcards and a "where" clause triggers empty_enum_arguments

### ISSUE THREAD

**Issue by fellipecaetano** [realm/SwiftLint#1722]
The following snippet is interpreted as a violation of the `empty_enum_arguments` rule:

```swift
switch (email, password) {
    case (_, _) where !email.value.isBlank && !password.value.isBlank:
        return .valid(email: .idle, password: .idle)
    default:
        break
}
```

I was expecting a violation of a different rule, or rather that no rules were violated at all.
